### PR TITLE
feat: dashboard table override

### PR DESCRIPTION
**Pull request by ktx-vaidehi**
- #4829 
- #4215 

## Release Notes

- **New Features**
	- Introduced the `OverrideConfigPopup` component for managing dashboard override configurations.
	- Added the `OverrideConfig` component for enhanced configuration options in table-type dashboards.
	- Enhanced `PanelConfig` to support override configurations.

- **Improvements**
	- Expanded dashboard panel data structure to include `override_config` for flexible configuration management.
	- Improved unit handling and date formatting in table data conversion.

These updates enhance customization and usability for dashboard configurations.

### CODE REVIEW

**Review by ktx-kirtan — APPROVED** [review]

**Comment by ktx-kirtan** [comment]
@coderabbitai review

File: web/src/components/dashboards/OverrideConfigPopup.vue [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+          v-model="overrideConfig.config[0].value.customUnit"
+          :label="t('dashboard.customunitLabel')"
+          color="input-border"
+          bg-color="input-bg"
+          stack-label
+          filled
+          dense
+          label-slot
+          data-test="dashboard-config-unit"
+        />
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/components/dashboards/OverrideConfigPopup.vue:69]: it's ok to not add it as required

File: web/src/components/dashboards/OverrideConfigPopup.vue [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+        JSON.stringify(
+          props.overrideConfig || [
+            {
+              field: { matchBy: "name", value: "" },
+              config: [{ type: "unit", value: { unit: "", customUnit: "" } }],
+            },
+          ],
+        ),
+      ),
+    );
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/components/dashboards/OverrideConfigPopup.vue:214]: for now it is not necessary to change

File: web/src/components/dashboards/addPanel/OverrideConfig.vue [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        <q-tooltip
+          class="bg-grey-8"
+          anchor="bottom middle"
+          self="top middle"
+          max-width="250px"
+        >
+          Map units to the selected columns for enhanced data visualization.
+        </q-tooltip>
+      </q-btn>
+    </div>
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/components/dashboards/addPanel/OverrideConfig.vue:22]: for now don't want this changes

File: web/src/components/dashboards/addPanel/OverrideConfig.vue [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    />
+
+    <q-dialog v-model="showOverrideConfigPopup">
+      <OverrideConfigPopup
+        :columns="columns"
+        :override-config="
+          JSON.parse(
+            JSON.stringify(dashboardPanelData.data.config.override_config),
+          )
+        "
```
[the remark this answers is not in the document]
Comment (ktx-vaidehi) [reply web/src/components/dashboards/addPanel/OverrideConfig.vue:39]: As we are using this structure of parse and stringify so for now this is ok can check later

**Pull request merged**